airflow/providers/alibaba/cloud/hooks/analyticdb_spark.py

```diff
@@ -0,0 +1,377 @@
+        executor_resource_spec: str | None = None,
+        num_executors: int | str | None = None,
+        name: str | None = None,
+    ) -> str:
+        """
+        Build the submit spark sql request data
+
+        :param sql: The SQL query to execute. (templated)
+        :param conf: Spark configuration properties.
+        :param driver_resource_spec: The resource specifications of the Spark driver.
+        :param executor_resource_spec: The resource specifications of each Spark executor.
+        :param num_executors: number of executors to launch for this application.
+        :param name: name of this application.
+        """
+        if sql is None:
+            raise Exception("Parameter sql is need when submit spark sql.")
+
+        extra_conf: dict[str, str] = {}
+        formatted_conf = ""
+
+        if driver_resource_spec:
+            extra_conf["spark.driver.resourceSpec"] = driver_resource_spec
+        if executor_resource_spec:
+            extra_conf["spark.executor.resourceSpec"] = executor_resource_spec
+        if num_executors:
+            extra_conf["spark.executor.instances"] = num_executors
+        if name:
+            extra_conf["spark.app.name"] = name
+        if conf and AnalyticDBSparkHook._validate_extra_conf(conf):
+            extra_conf.update(conf)
+        for key, value in extra_conf.items():
+            formatted_conf += f'set {key} = {value};'
+
+        return (formatted_conf + sql).strip()
+
+    @staticmethod
+    def _validate_list_of_stringables(vals: Sequence[str | int | float]) -> bool:
+        """
+        Check the values in the provided list can be converted to strings.
+
+        :param vals: list to validate
+        :return: true if valid
+        """
+        if (
+            vals is None
+            or not isinstance(vals, (tuple, list))
+            or any(1 for val in vals if not isinstance(val, (str, int, float)))
+        ):
+            raise ValueError("List of strings expected")
+        return True
+
+    @staticmethod
+    def _validate_extra_conf(conf: dict[Any, Any]) -> bool:
+        """
+        Check configuration values are either strings or ints.
+
+        :param conf: configuration variable
+        :return: true if valid
+        """
+        if conf:
+            if not isinstance(conf, dict):
+                raise ValueError("'conf' argument must be a dict")
+            if any(True for k, v in conf.items() if not (v and isinstance(v, str) or isinstance(v, int))):
+                raise ValueError("'conf' values must be either strings or ints")
+        return True
+
+    def get_adb_spark_client(self) -> Client:
+        """
+        Get valid AnalyticDB MySQL Spark client
+        """
+        assert self.region is not None
+
+        extra_config = self.adb_spark_conn.extra_dejson
+        auth_type = extra_config.get("auth_type", None)
+        if not auth_type:
+            raise Exception("No auth_type specified in extra_config.")
```

Comment:
Maybe a more specific exception type will be nice, so that code handling it doesn't have to use `except Exception` which can potentially mask all other errors.

Reply:
I think replacing every instance of `raise Exception` would be nice.

Reply:
Agreed with you.   
Replace Exception with ValueError. I summarize the scenarios where Exception is used:
1. Spark application parameters specified by user;
2. Connection content;